File: app/assets/javascripts/EnteralFormula.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const EnteralFormulaSchema = DataElementSchema({
  baseFormulaType: Any,
  baseFormulaProductName: Any,
  additiveType: Any,
  additiveProductName: Any,
  caloricDensity: Any,
  routeofAdministration: Any,
  administration: Any,
  maxVolumeToDeliver: Any,
  administrationInstruction: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'EnteralFormula' },

});

module.exports.EnteralFormulaSchema = EnteralFormulaSchema;
module.exports.EnteralFormula = mongoose.model('EnteralFormula', EnteralFormulaSchema);

File: app/models/qdm/enteral_formula.rb
module QDM
  # app/models/qdm/enteral_formula.rb
  class EnteralFormula < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :baseFormulaType
    field :baseFormulaProductName
    field :additiveType
    field :additiveProductName
    field :caloricDensity
    field :routeofAdministration
    field :administration
    field :maxVolumeToDeliver
    field :administrationInstruction
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
